### PR TITLE
feat(sdk)!: provide request execution information

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

In SDK we would like to have more information on request execution including address and retries count for logging, error handling, and retry implementation on the SDK level

## What was done?

- Added `ExecutionResult` type that contains `ExecutionResponse` and `ExecutionError`. Both stucts contains `retries`, `address` and inner objects.
- Updated retry policy to make delays between retries constant. Since we are doing requests to different nodes there is no point in having exponential delays.
- Removed `Address` from the `DapiClientError::Transport` error
- Added temporary handlers for `ExecutionResponse` and `ExecutionError` to SDK 

## How Has This Been Tested?

Running tests

## Breaking Changes

`DapiRequestExecutor::execute` and `DapiRequest::execute` now returns `ExecutionResult` 

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling and response processing for various methods within the SDK, improving clarity and robustness.
	- Introduction of new `ExecutionResult` type for better handling of execution contexts in requests.
	- New JSON files added for testing various functionalities related to quorum public keys and identity management.

- **Bug Fixes**
	- Removed deprecated files and improved error handling across multiple modules.

- **Documentation**
	- Updated comments and documentation to reflect changes in error handling and new features.

### CODE REVIEW

File: packages/rs-dapi-client/src/transport.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ pub trait TransportRequest: Clone + Send + Sync + Debug + Mockable {
 /// Generic way to create a transport client from provided [Uri].
 pub trait TransportClient: Send + Sized {
     /// Error type for the specific client.
-    type Error: CanRetry + Send + Debug + Mockable;
+    type Error: std::error::Error + CanRetry + Send + Mockable;
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-dapi-client/src/transport.rs:51]: Don't worry, we will use only DAPI Client errors here

File: packages/rs-dapi-client/src/dapi_client.rs [hunk 2]
```diff
@@ -140,24 +124,25 @@ impl DapiRequestExecutor for DapiClient {
...
 
         // Save dump dir for later use, as self is moved into routine
         #[cfg(feature = "dump")]
         let dump_dir = self.dump_dir.clone();
         #[cfg(feature = "dump")]
         let dump_request = request.clone();
 
+        let retries_counter_arc = Arc::new(AtomicUsize::new(0));
+        let retries_counter_arc_ref = &retries_counter_arc;
+
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-dapi-client/src/dapi_client.rs:139]: If you see we pass it inside asynchronous code so it should Send + Sync.

File: packages/rs-sdk/src/platform/transition/update_price_of_document.rs [hunk 3]
```diff
@@ -112,7 +113,11 @@
     ) -> Result<Document, Error> {
         let request = state_transition.wait_for_state_transition_result_request()?;
 
-        let response = request.execute(sdk, RequestSettings::default()).await?;
+        let response = request
+            .execute(sdk, RequestSettings::default())
+            .await // TODO: We need better way to handle execution response and errors
+            .map(|execution_response| execution_response.into_inner())
+            .map_err(|execution_error| execution_error.into_inner())?;
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-sdk/src/platform/transition/update_price_of_document.rs:120]: Fixed in upcoming PRs

File: packages/rs-dapi-client/src/executor.rs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+/// Request execution response.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ExecutionResponse<R> {
+    /// The response from the request
+    pub inner: R,
+    /// How many times the request was retried
+    pub retries: usize,
+    /// The address of the node that was used for the request
+    pub address: Address,
+}
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-dapi-client/src/executor.rs:56]: We always have address if we have response

File: packages/rs-dapi-client/src/lib.rs [hunk 5]
```diff
@@ -48,7 +49,7 @@
         self,
         dapi_client: &'c D,
         settings: RequestSettings,
-    ) -> BoxFuture<'c, Result<Self::Response, DapiClientError<Self::TransportError>>>
+    ) -> BoxFuture<'c, ExecutionResult<Self::Response, DapiClientError<Self::TransportError>>>
     where
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-dapi-client/src/lib.rs:53]: Yes, this PR marked as a breaking change

**Review by lklimek — APPROVED** [review]

**Pull request merged**